streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalProcessorContextImpl.java

```diff
@@ -110,6 +112,11 @@ public long currentStreamTimeMs() {
         throw new UnsupportedOperationException("There is no concept of stream-time for a global processor.");
     }
 
+    @Override
+    public Map<TopicPartition, Long> currentPositions() {
+        throw new UnsupportedOperationException("currentPositions is not supported for global processors.");
```

Comment:
Why not? (We also support `currentSystemTimeMs()` -- we only don't support `currentStreamTimeMs()` because the global thread processed independently of the main threads, and there is no concept of stream-time for the global task.

Reply:
Two reasons:
1. The code isn't already in place to track it (because we don't commit global tasks), and I'm lazy :)
2. I think the reasoning is similar to the currentStreamTimeMs: the Global thread doesn't have a coherent "task" that has a "stream time" or a "position". It's more similar to the restore consumer. It's just a bunch of partitions that all got grouped together for batching efficiency.  It seems like there could be a use case for asking the current position of the global thread (or the restore consumers) on a particular partition, but it's not the same semantics as finding out the "current position of this task on its inputs".

So, I don't think it would be invalid to add it, but it's both not that easy and different from what I'm trying to support here. I'll mention this in the KIP as a "deferred for now" option.

Reply:
I agree that it might not be required (useful?) to support. However, I think it's different to "stream-time" because (1) stream-time depends on ordered processing across partitions of different topics (what does not apply to a global task that reads a single topic and multiple/all partitions of this single topic) and (2) a global-task is _decoupled_ from the other tasks for "time synchronization" (time synchronization is important for joins!). -- However, for just tracking the offset per partition, there is no cross-task synchronization necessary and we can report the offset per partition easily. Thus, it's a well-define operation for a global-task too, while "stream-time" is undefined for a global-task.

> The code isn't already in place to track it (because we don't commit global tasks)

Don't think it would be hard to add? Should even be simpler than for regular tasks, because we don't have a "record buffer", but only the current `ConsumerRecord` iterator: so we can just do a look-ahead of one record, or use `position()` if we are at the end of the iterator.